src/mixins/nft.js

```diff
@@ -271,6 +274,35 @@ export default {
         ? this.listingInfo.price
         : this.purchaseInfo.price;
     },
+    isCollectExpiryTimeNear() {
+      const { collectExpiryAt } = this.purchaseInfo;
+      if (!collectExpiryAt) return false;
+
+      const timeLeft = collectExpiryAt - Date.now();
+      const threshold = 4 * 7 * 24 * 60 * 60 * 1000; // 4 weeks
+      return timeLeft > 0 && timeLeft < threshold;
+    },
+    collectExpiryTime() {
+      const { collectExpiryAt } = this.purchaseInfo;
+      if (!collectExpiryAt) return '';
+
+      const timeLeft = collectExpiryAt - Date.now();
+      if (timeLeft <= 0) return '';
+
+      const dateTime = DateTime.fromMillis(collectExpiryAt);
+      if (this.isCollectExpiryTimeNear) {
+        const duration = dateTime
+          .toRelative({ locale: this.$i18n.locale })
+          .replace('in ', '') // en
+          .replace('後', ''); // zh
+        return this.$t('nft_collect_expiry_time_near', { duration });
+      }
+      const date = dateTime.toFormat('yyyy-MM-dd');
+      return this.$t('nft_collect_expiry_time_far', { date });
+    },
+    collectExpiryTimeForNear() {
+      return this.isCollectExpiryTimeNear ? this.collectExpiryTime : '';
+    },
```

Comment:
```suggestion
    collectExpiryTimeForDisplay() {
      return this.isCollectExpiryTimeNear ? this.collectExpiryTime : '';
    },
```

Reply:
Since `collectExpiryTime` field is also for display, I renamed `collectExpiryTimeForDisplay` to `collectExpiryTimeForHighlight`